test/join.test.ts

```diff
@@ -16,6 +16,10 @@ const joinURLTests = [
   { input: ["/", "./foo"], out: "/foo" },
   { input: ["/", "./foo/"], out: "/foo/" },
   { input: ["/", "./foo", "bar"], out: "/foo/bar" },
+  {
+    input: ["https://google.com/", "./foo", "/bar"],
+    out: "https://google.com/foo/bar",
+  },
```

Comment:
Ideally we also should not ascend above the URL in this case:

```suggestion
  {
    input: ["https://google.com/", "./foo", "/bar"],
    out: "https://google.com/foo/bar",
  },
  {
    input: ["https://google.com/", "../foo"],
    out: "https://google.com/foo",
  },
```

Reply:
he we go with edge cases 😆